src/features/query-bar/components/filter/FilterButton.js

```diff
@@ -41,30 +42,27 @@ class FilterButton extends React.Component<Props, State> {
         super(props);
 
         this.state = {
-            transientConditions: this.props.conditions.slice(0),
-            selectedConnector: AND,
             areErrorsEnabled: false,
             isMenuOpen: false,
+            selectedConnector: AND,
+            transientConditions: cloneDeep(this.props.conditions),
         };
     }
 
-    componentDidMount() {
-        const initialCondition = this.createCondition();
-
-        this.setState({
-            transientConditions: [initialCondition],
-        });
-    }
-
-    componentDidUpdate() {
-        const { conditions } = this.props;
-        const { transientConditions } = this.state;
-        const wasFlyoutClosed = transientConditions.length === 0;
-        const shouldHydrateConditions = transientConditions.length !== conditions.length;
-        if (wasFlyoutClosed && shouldHydrateConditions) {
-            this.setState({
-                transientConditions: this.props.conditions.slice(0),
-            });
+    componentDidUpdate(prevProps: Props, prevState: State) {
+        const { isMenuOpen } = this.state;
+        const { isMenuOpen: prevIsMenuOpen } = prevState;
+        const wasFlyoutOpened = isMenuOpen && !prevIsMenuOpen;
+        if (wasFlyoutOpened) {
+            if (this.props.conditions.length === 0) {
+                this.setState({
+                    transientConditions: [this.createCondition()],
```

Comment:
if empty condition
1. and empty columns, then transientC = []
1. and non-empty columns, then transientC = [{...}]

In createCondition(), if no columns then throw Error('Columns required');